nodejs/week1/mealSharingWebsite/src/backend/index.js

```diff
@@ -0,0 +1,36 @@
+const express = require("express");
+const app = express();
+
+app.use((req, res, next) => {
+    console.log("method " + req.method);
+    next();
+});
+app.use((req, res, next) => {
+    console.log("date " + Date.now());
+    next();
+});
+
+app.use("/", (req, res) => res.send("Error no page found"));
+
+
+const mealRouter = require("./routes/meals.js");
+app.use("/meals", mealRouter);
+
+const cheapMealRouter = require("./routes/cheapMeals.js");
+app.use("/cheap-meals", cheapMealRouter);
+
+const largeMealRouter = require("./routes/largeMeals.js");
+app.use("/large-meals", largeMealRouter);
+
+const randomMealRouter = require("./routes/meal.js");
+app.use("/meal", randomMealRouter);
+
+const reservationsRouter = require("./routes/reservations.js");
+app.use("/reservations", reservationsRouter);
+
+const randomReservationsRouter = require("./routes/reservation.js");
+app.use("/reservation", randomReservationsRouter);
```

Comment:
@gizemcandemir [app.get](http://expressjs.com/en/api.html#app.get) is meant to process only HTTP GET requests whereas [app.use](http://expressjs.com/en/api.html#app.use) can be used to handle all types of HTTP requests.
Although in this homework assignment, all the HTTP requests are for the GET method, `app.use` here is still better since it allows you to add more methods in the respective routers for future homework.
Please let me know if what I stated above is unclear.

Reply:
Thanks @jawkha ! 
Now I understood, and it totally makes sense 👍🏼